### PR TITLE
Add error return codes to h3Index.c (part 1)

### PR DESCRIPTION
**Pull request by isaacbrodsky**
Applies the error return codes RFC to h3Index.c with the exception of some commonly used functions that return true/false.

### CODE REVIEW

**Review by dfellis — APPROVED** [review]

File: src/apps/testapps/testCellToChildren.c [hunk 1]
```diff
@@ -63,11 +63,17 @@ static void assertSetsEqual(H3Index *set1, int len1, H3Index *set2, int len2) {
...
-    int64_t numChildren = H3_EXPORT(cellToChildrenSize)(h, res);
+static void checkChildren(H3Index h, int res, H3Error expectedError,
+                          H3Index *expected, int numExpected) {
+    int64_t numChildren;
+    H3Error numChildrenError =
+        H3_EXPORT(cellToChildrenSize)(h, res, &numChildren);
+    t_assert(numChildrenError == expectedError, "Expected error code");
+    if (expectedError != E_SUCCESS) {
+        return;
+    }
```
Comment (nrabinowitz) [inline src/apps/testapps/testCellToChildren.c:74]: Does this imply that there are no errors we can test that won't be returned by `cellToChildrenSize`? I.e. there are no errors in `cellToChildren` that aren't in `cellToChildrenSize`?

File: src/apps/testapps/testH3Index.c [hunk 2]
```diff
@@ -168,24 +168,26 @@ SUITE(h3Index) {
...
 
     TEST(stringToH3) {
-        t_assert(H3_EXPORT(stringToH3)("") == 0, "got an index from nothing");
-        t_assert(H3_EXPORT(stringToH3)("**") == 0, "got an index from junk");
-        t_assert(
-            H3_EXPORT(stringToH3)("ffffffffffffffff") == 0xffffffffffffffff,
-            "failed on large input");
+        H3Index h3;
+        t_assert(H3_EXPORT(stringToH3)("", &h3) == E_FAILED,
+                 "got an index from nothing");
```
Comment (nrabinowitz) [inline src/apps/testapps/testH3Index.c:186]: Nit:
```suggestion
                 "no index from nothing");
```

File: src/apps/testapps/testH3Index.c [hunk 3]
```diff
@@ -168,24 +168,26 @@ SUITE(h3Index) {
...
+        t_assert(H3_EXPORT(stringToH3)("**", &h3) == E_FAILED,
+                 "got an index from junk");
```
Comment (nrabinowitz) [inline src/apps/testapps/testH3Index.c:188]:
```suggestion
                 "no index from junk");
```

File: src/apps/testapps/testH3Index.c [hunk 4]
```diff
@@ -168,24 +168,26 @@ SUITE(h3Index) {
...
+        t_assertSuccess(H3_EXPORT(stringToH3)("ffffffffffffffff", &h3));
+        t_assert(h3 == 0xffffffffffffffff, "failed on large input");
```
Comment (nrabinowitz) [inline src/apps/testapps/testH3Index.c:190]:
```suggestion
        t_assert(h3 == 0xffffffffffffffff, "got expected on large input");
```
Clearly we have some inconsistencies on how we write our assertion messages (I usually prefer the positive "got expected" or the prescriptive "must get expected"; I don't really like the negative "here's what went wrong", as it doesn't explain expected behavior well)

**Review by nrabinowitz — APPROVED** [review]

File: src/apps/testapps/testCellToChildren.c (hunk 1 above)
Comment (isaacbrodsky) [reply src/apps/testapps/testCellToChildren.c:74]: `cellToChildren` always returns `E_SUCCESS` right now, so that is correct

**Pull request merged**